pkg/db/pgx.go

```diff
@@ -135,44 +135,90 @@ func createNamespace(
 	return nil
 }
 
-// NewNamespacedDB creates a new database with the given namespace if it doesn't exist and returns the full DSN for the new database.
-func NewNamespacedDB(
+type dbConnConfig struct {
+	pingTimeout        time.Duration
+	statementTimeout   time.Duration
+	prometheusRegistry *prometheus.Registry
+	createNamespace    bool
+	runMigrations      bool
+}
+
+type dbOptionFunc func(*dbConnConfig)
+
+func dbPingTimeout(d time.Duration) dbOptionFunc {
+	return func(cfg *dbConnConfig) {
+		cfg.pingTimeout = d
+	}
+}
+
+func dbStatementTimeout(d time.Duration) dbOptionFunc {
+	return func(cfg *dbConnConfig) {
+		cfg.statementTimeout = d
+	}
+}
+
+// doCreateNamespace will create the namespace if it does not exist.
+func doCreateNamespace(b bool) dbOptionFunc {
+	return func(cfg *dbConnConfig) {
+		cfg.createNamespace = b
+	}
+}
+
+func runMigrations(b bool) dbOptionFunc {
+	return func(cfg *dbConnConfig) {
+		cfg.runMigrations = b
+	}
+}
+
+func prometheusRegistry(p *prometheus.Registry) dbOptionFunc {
+	return func(cfg *dbConnConfig) {
+		cfg.prometheusRegistry = p
+	}
+}
+
+func connectToDB(
```

Comment:
This function is introduced as a way to unify the two variants previously present. There were two almost identical functions, one that creates the namespace and runs migrations and one that does neither of those.

Now the function is generalized, but the external interface of the package did not change.

There is one change right now though, previously the  `NewNamespacedDB` did:

```go
config.ConnConfig.Database = namespace
```

while `ConnectToDB` did a conditional overwrite:

```go
if namespace != "" {
	config.ConnConfig.Database = namespace
}
```

The difference seems inconsequential but wanted to flag it.